conceptnet5/vectors/transforms.py

```diff
@@ -101,3 +148,72 @@ def make_replacements(small_frame, big_frame, limit=100000):
             if got:
                 replacements[term] = got[0]
     return replacements
+
+
+def choose_small_vocabulary(big_frame):
+    """
+    Choose the vocabulary of the small frame, by eliminating the terms which:
+     - contain more than one word
+     - are not in ConceptNet
+    """
+    small_vocabulary = []
+    finder = AssertionFinder()
+
+    for term in big_frame.index:
+        # Make sure the term is not a phrase
+        if term.count('_') < 1:
+            # Check if a term comes from ConceptNet
+            results = finder.lookup(term)
```

Comment:
This seems like it's going to be very slow, and also unnecessarily depend on the DB being built, when it could happen in parallel with building the DB.

You could instead take `data/stats/core_concepts.txt` as an input and just read those URIs into a set.